### PR TITLE
Fat 5314 create dal

### PR DESCRIPTION
**Pull request by Activict**
Contains development DAL

### CODE REVIEW

File: StoreApp.DAL/EF/ModelContext.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+﻿using System.Data.Entity;
+using StoreApp.DAL.Entities;
+
+namespace StoreApp.DAL.EF
+{
+    public class StoreContext : DbContext
```
Comment (AleksandroBoko) [inline StoreApp.DAL/EF/ModelContext.cs]: Need to correct class name. The name of the file and name of the class should be the same

File: StoreApp.DAL/Entities/Order.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace StoreApp.DAL.Entities
+{
+    public class Order
+    {
+        public int Id { get; set; }
+        public int? UserId { get; set; }
```
Comment (AleksandroBoko) [inline StoreApp.DAL/Entities/Order.cs]: It seems that you expect to have NULL here but I am not sure that we need to have an order without User

File: StoreApp.DAL/Entities/OrderDetail.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿namespace StoreApp.DAL.Entities
+{
+    public class OrderDetail
+    {
+        public int Id { get; set; }
+        public int? OrderId { get; set; }
```
Comment (AleksandroBoko) [inline StoreApp.DAL/Entities/OrderDetail.cs]: The same as before

File: StoreApp.DAL/Entities/OrderDetail.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+        public int? ProductId { get; set; }
```
Comment (AleksandroBoko) [inline StoreApp.DAL/Entities/OrderDetail.cs]: The same as before

File: StoreApp.DAL/Entities/Order.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        public string Status { get; set; }
+        public int Discount { get; set; }
+        public DateTime OrderDate { get; set; }
+
+        public virtual User User { get; set; }
+
+        public virtual ICollection<OrderDetail> OrderDetails { get; set; }
+        public Order()
+        {
+            OrderDetails = new List<OrderDetail>();
```
Comment (AleksandroBoko) [inline StoreApp.DAL/Entities/Order.cs]: Why do we need to initialize this collection of details?

File: StoreApp.DAL/Entities/Producer.cs [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+{
+    public class Producer
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+
+        public virtual ICollection<Product> Products { get; set; }
+        public Producer()
+        {
+            Products = new List<Product>();
```
Comment (AleksandroBoko) [inline StoreApp.DAL/Entities/Producer.cs]: The same as before

File: StoreApp.DAL/Entities/Product.cs [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        public int? ProducerId { get; set; }
+
+        public virtual Category Category { get; set; }
+        public virtual Brand Brand { get; set; }
+        public virtual Producer Producer { get; set; }
+
+        public virtual ICollection<OrderDetail> OrderDetails { get; set; }
+        public Product()
+        {
+            OrderDetails = new List<OrderDetail>();
```
Comment (AleksandroBoko) [inline StoreApp.DAL/Entities/Product.cs]: The same as before

File: StoreApp.DAL/Entities/Product.cs [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public decimal Price { get; set; }
+        public int Quantity { get; set; }
+        public string Unit { get; set; }
+        public string Picture { get; set; }
+        public string Quality { get; set; }
+        public bool Enable { get; set; }
+
+        public int? CategoryId { get; set; }
```
Comment (AleksandroBoko) [inline StoreApp.DAL/Entities/Product.cs]: The same as before

File: StoreApp.DAL/Entities/Product.cs [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+        public int? BrandId { get; set; }
```
Comment (AleksandroBoko) [inline StoreApp.DAL/Entities/Product.cs]: The same as before

File: StoreApp.DAL/Entities/Product.cs [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+        public int? ProducerId { get; set; }
```
Comment (AleksandroBoko) [inline StoreApp.DAL/Entities/Product.cs]: The same as before

File: StoreApp.DAL/Entities/User.cs [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+        public int Id { get; set; }
+        public string Email { get; set; }
+        public string Password { get; set; }
+        public string UserName { get; set; }
+        public string HomeAdress { get; set; }
+
+        public virtual ICollection<Order> Orders { get; set; }
+        public User()
+        {
+            Orders = new List<Order>();
```
Comment (AleksandroBoko) [inline StoreApp.DAL/Entities/User.cs]: The same as before

File: StoreApp.DAL/Repositories/BrandRepository.cs [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+using System.Collections.Generic;
+
+namespace StoreApp.DAL.Repositories
+{
+    class BrandRepository : IRepository<Brand>
+    {
+        private StoreContext db;
+        public BrandRepository(StoreContext context)
+        {
+            this.db = context;
```
Comment (AleksandroBoko) [inline StoreApp.DAL/Repositories/BrandRepository.cs]: It is not necessary to use this.

File: StoreApp.DAL/Repositories/CategoryRepository.cs [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+using System.Collections.Generic;
+
+namespace StoreApp.DAL.Repositories
+{
+    class CategoryRepository : IRepository<Category>
+    {
+        private StoreContext db;
+        public CategoryRepository(StoreContext context)
+        {
+            this.db = context;
```
Comment (AleksandroBoko) [inline StoreApp.DAL/Repositories/CategoryRepository.cs]: The same as before

File: StoreApp.DAL/EF/ModelContext.cs (hunk 1 above)
Comment (Activict) [reply StoreApp.DAL/EF/ModelContext.cs]: Agree, I will fix it

File: StoreApp.DAL/Entities/Order.cs (hunk 2 above)
Comment (Activict) [reply StoreApp.DAL/Entities/Order.cs]: Agree, I will fix it

File: StoreApp.DAL/Entities/OrderDetail.cs (hunk 3 above)
Comment (Activict) [reply StoreApp.DAL/Entities/OrderDetail.cs]: Agree, I will fix it

File: StoreApp.DAL/Entities/OrderDetail.cs (hunk 4 above)
Comment (Activict) [reply StoreApp.DAL/Entities/OrderDetail.cs]: Agree, I will fix it

File: StoreApp.DAL/Entities/Order.cs (hunk 5 above)
Comment (Activict) [reply StoreApp.DAL/Entities/Order.cs]: Agree, I will fix it

File: StoreApp.DAL/Entities/Producer.cs (hunk 6 above)
Comment (Activict) [reply StoreApp.DAL/Entities/Producer.cs]: It constructor need for Lazy loading

File: StoreApp.DAL/Entities/Product.cs (hunk 7 above)
Comment (Activict) [reply StoreApp.DAL/Entities/Product.cs]: It constructor need for Lazy loading

File: StoreApp.DAL/Entities/Product.cs (hunk 8 above)
Comment (Activict) [reply StoreApp.DAL/Entities/Product.cs]: Agree, I will fix it

File: StoreApp.DAL/Entities/Product.cs (hunk 9 above)
Comment (Activict) [reply StoreApp.DAL/Entities/Product.cs]: Agree, I will fix it

File: StoreApp.DAL/Entities/Product.cs (hunk 10 above)
Comment (Activict) [reply StoreApp.DAL/Entities/Product.cs]: Agree, I will fix it

File: StoreApp.DAL/Entities/User.cs (hunk 11 above)
Comment (Activict) [reply StoreApp.DAL/Entities/User.cs]: Agree, I will fix it

File: StoreApp.DAL/Repositories/BrandRepository.cs (hunk 12 above)
Comment (Activict) [reply StoreApp.DAL/Repositories/BrandRepository.cs]: Agree, I will fix it

File: StoreApp.DAL/Repositories/CategoryRepository.cs (hunk 13 above)
Comment (Activict) [reply StoreApp.DAL/Repositories/CategoryRepository.cs]: Agree, I will fix it

**Pull request merged**